### PR TITLE
Implement GPI2 gate in transpiler

### LINKED ISSUE #548: Missing GPI2 decomposition
In https://github.com/qiboteam/qibocal/pull/452 I'm trying to run a circuit on hardware containing the `GPI2` gate implemented in Qibo and I got the following error.
```sh
KeyError: <class 'qibo.gates.gates.GPI2'>
````
If I understood correctly the reason for this is that we are missing the GPI2 gate in the [gate decomposition](https://github.com/qiboteam/qibolab/blob/main/src/qibolab/transpilers/gate_decompositions.py) the transpiler.

Actually it would nice to make sure that all gate represented in Qibo have their decomposition also in the transpiler.

### PR DESCRIPTION
**Pull request by andrea-pasquale**
Closes #548. 
I don't have a specific reference but according to what I saw in https://github.com/qiboteam/qibocal/pull/452 it should be correct.

Checklist:
- [ ] Reviewers confirm new code works as expected.
- [x] Tests are passing.
- [x] Coverage does not decrease.
- [x] Documentation is updated.

### CODE REVIEW

File: src/qibolab/compilers/default.py [hunk 1]
```diff
@@ -24,6 +24,17 @@ def rz_rule(gate, platform):
...
 
+def gpi2_rule(gate, platform):
+    """Rule for GPI2."""
+    qubit = gate.target_qubits[0]
+    theta = gate.parameters[0]
+
+    sequence = PulseSequence()
+    pulse = platform.create_RX90_pulse(qubit, start=0, relative_phase=theta)
+    sequence.add(pulse)
+    return sequence, {qubit: gate.parameters[0]}
```
Comment (stavros11) [inline src/qibolab/compilers/default.py:35]: I am not sure if GPI2 requires to add a virtual phase in the upcoming pulses, which is what this dictionary is for. I have the impression that you only need to put `theta` in the RX90 pulse phase, which you do, and there is no further virtual-Z rotation, but I am not sure at all so it is better if @igres26 can confirm this.

**Review by stavros11 — COMMENTED** [review]
Thanks for implementing this @andrea-pasquale, looks good to me, we just need to check the point below.

Now that we have this, we can also modify the native gate transpiler to transpile one-qubit gates to RZ and GPI2, instead of RZ and U3 like we do now. This will probably result to more efficient translations in terms of pulses. Better to do this in a different PR though.

**Comment by andrea-pasquale** [comment]
> Thanks for implementing this @andrea-pasquale, looks good to me, we just need to check the point below.
> 
> Now that we have this, we can also modify the native gate transpiler to transpile one-qubit gates to RZ and GPI2, instead of RZ and U3 like we do now. This will probably result to more efficient translations in terms of pulses. Better to do this in a different PR though.

Thanks @stavros11 for the review.
Indeed, I try to modify some tests to fix the coverage but I saw that there were some errors (this is why I converted this PR to draft). As you suggested it is better to do this in a separate PR.

File: src/qibolab/compilers/default.py (hunk 1 above)
Comment (andrea-pasquale) [reply src/qibolab/compilers/default.py:35]: I see, I don't think so. Lets wait for @igres26 feedback.

**Comment by stavros11** [comment]
> Indeed, I try to modify some tests to fix the coverage but I saw that there were some errors (this is why I converted this PR to draft). As you suggested it is better to do this in a separate PR.

About coverage, I think you need to add a test that triggers the compilation rule, something like the tests in test_compilers_default.py.

File: src/qibolab/compilers/default.py (hunk 1 above)
Comment (andrea-pasquale) [reply src/qibolab/compilers/default.py]: I had a chat with @igres26 and there is no further virtual-Z rotation need. I've removed it in https://github.com/qiboteam/qibolab/pull/549/commits/7620487155e9443c8f0f9d57c1f79b25898b0f77.

**Review by stavros11 — APPROVED** [review]
Thanks, sounds good. 

By the way, I believe the GPI gate, which I think is only a virtual-Z phase, is also missing here. But the same is true for other qibo gates introduced recently so maybe we can merge this that is blocking other developments and add the rest later.

**Comment by andrea-pasquale** [comment]
Indeed you are right @stavros11.
For what I am doing now I only need the GPI2.
I will open an issue to make sure that we port all the gates available in Qibo also here in Qibolab.

**Review by rodolfocarobene — APPROVED** [review]
Thanks @andrea-pasquale, everything looks good to me

**Pull request merged**